src/screens/diagramTool/DiagramTool.js

```diff
@@ -1,155 +1,190 @@
-    const windowHeight = Dimensions.get('window').height - (Dimensions.get('window').height/4);
+  const { navigation, addPhoto, photo } = props;
+  const drawRef = useRef(null);
 
-    const [color, setColor] = useState(DEFAULT_COLORS[1][0][0]);
-    const [thickness, setThickness] = useState(5);
-    const [opacity, setOpacity] = useState(1);
-    const [tool, setTool] = useState(DrawingTool.Brush);
-    const [visibleBrushProperties, setVisibleBrushProperties] = useState(false);
-    const [alert, setAlert] = useState(false);
+  // Get the dimensions of device screen to properly place drawing canvas on screen
+  const windowWidth = Dimensions.get("window").width;
+  const windowHeight =
+    Dimensions.get("window").height - Dimensions.get("window").height / 4;
 
-    const handleClose = () => {
-        setAlert(false)
-    }
+  // Set default values for drawing tool parameters
+  const [color, setColor] = useState(DEFAULT_COLORS[1][0][0]);
+  const [thickness, setThickness] = useState(5);
+  const [opacity, setOpacity] = useState(1);
+  const [tool, setTool] = useState(DrawingTool.Brush);
+  const [visibleBrushProperties, setVisibleBrushProperties] = useState(false);
+  const [alert, setAlert] = useState(false);
 
-    const handleUndo = () => {
-     drawRef.current.undo();
-    };
+  const handleClose = () => {
+    // close the alert dialog
+    setAlert(false);
+  };
 
-    const handleClear = () => {
-        if (!alert) {
-            setAlert(true)
-        } else {
-            drawRef.current.clear();
-            setAlert(false);
-        }
-    };
+  const handleUndo = () => {
+    // undos the last line drawn
+    drawRef.current.undo();
+  };
 
-    const handleToggleEraser = () => {
-     setTool((prev) =>
-       prev === DrawingTool.Brush ? DrawingTool.Eraser : DrawingTool.Brush
-     );
-    };
+  const handleClear = () => {
+    // prompts alert with delete warning to appear before user can delete drawing
+    if (!alert) {
+      setAlert(true);
+    } else {
+      // deletes all paths of drawing
+      drawRef.current.clear();
+      setAlert(false);
+    }
+  };
 
-    const [overlayOpacity] = useState(new Animated.Value(0));
-    const handleToggleBrushProperties = () => {
-     if (!visibleBrushProperties) {
-       setVisibleBrushProperties(true);
+  const handleToggleEraser = () => {
+    // sets the user tool as brush or eraser
+    setTool((prev) =>
+      prev === DrawingTool.Brush ? DrawingTool.Eraser : DrawingTool.Brush
+    );
+  };
 
-       Animated.timing(overlayOpacity, {
-         toValue: 1,
-         duration: 200,
-         useNativeDriver: true,
-       }).start();
-     } else {
-       Animated.timing(overlayOpacity, {
-         toValue: 0,
-         duration: 200,
-         useNativeDriver: true,
-       }).start(() => {
-         setVisibleBrushProperties(false);
-       });
-     }
+  const [overlayOpacity] = useState(new Animated.Value(0));
```

Comment:
could this be a local variable inside the `handleToggleBrushProperties` function, or does it need to be accessed outside the function?

Reply:
It is needed outside of the function